src/components/GlobalPreferences/Network/Network.js

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import styled from 'styled-components'
+import {
+  Box,
+  Button,
+  Info,
+  GU,
+  TextInput,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { defaultEthNode, defaultIpfsGateway } from '../../../networks'
+import { setDefaultEthNode, setIpfsGateway } from '../../../local-settings'
+import { InvalidNetworkType, InvalidURI, NoConnection } from '../../../errors'
+import keycodes from '../../../keycodes'
+import {
+  checkValidEthNode,
+  getNetworkType,
+  sanitizeNetworkType,
+} from '../../../lib/web3-utils'
+
+function Network() {
+  const {
+    ethNode,
+    networkType,
+    ipfsGateway,
+    handleEthNodeChange,
+    handleIpfsGatewayChange,
+    handleClearCache,
+    networkError,
+    handleNetworkChange,
+  } = useNetwork()
+  const theme = useTheme()
+
+  const { layoutName } = useLayout()
+  const compact = layoutName === 'small'
+
+  return (
+    <React.Fragment>
+      <Box heading="Node settings">
```

Comment:
We should add a setting for the subgraph API endpoint (and like the ETH node, do some sanity checks against it to make sure it works).

Reply:
just to undertand why do we want to let the user change the data source of the dashboard?

Reply:
just to understand why do we want to let the user change the data source of the dashboard?

Reply:
In case they wanted to run their own hosted instance of the subgraph, just like the ETH node or IPFS gateway.

Reply:
https://github.com/aragon/court-dashboard/issues/273